src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -134,10 +134,19 @@ public static string GetFullPath(string path)
                 throw new PathTooLongException(SR.IO_PathTooLong);
             }
 
-            return
-                sb.Length == 0 ? (fullCheck ? DirectorySeparatorCharAsString : string.Empty) :
-                sb.Length == path.Length ? path :
-                sb.ToString();
+            if (sb.Length == 0)
+            {
+                return fullCheck ? DirectorySeparatorCharAsString : string.Empty;
+            }
+            else if (sb.Length == path.Length)
+            {
+                StringBuilderCache.Release(sb);
+                return path;
+            }
+            else
+            {
+                return StringBuilderCache.GetStringAndRelease(sb);
+            }
```

Comment:
Why isn't this releasing the StringBuilder in the first case, too?  Rather than rewrite the whole expression with multiple calls to GetStringAndRelease, rather than the current code that does:
```C#
return stuff;
```
I'd do:
```C#
string result = stuff;
StringBuilderCache.Release(sb);
return result;
```